### DISCUSSION TITLE
How to pass some "active content" (or callback function) to partials?

### DISCUSSION THREAD

**Discussion by cagix** [McShelby/hugo-theme-relearn#504] [Q&A] [answered]
I'd like to pass something like (actually defined as partial)

```
{{ $quizzes := .Params.quizzes }}

<ul>
{{ range $quizzes }}
    {{ $name := index . "name" }}
    {{ $link := index . "link" }}
    <li><a href="{{- $link -}}">{{- default $link $name -}}</a></li>
{{ end }}
</ul>
```

as `content` parameter to your partial [`shortcodes/notice.html`](https://github.com/McShelby/hugo-theme-relearn/blob/main/layouts/partials/shortcodes/notice.html). Is there any way to achieve something like this?

**Comment by McShelby** [comment, marked as answer]
Sure. Basically:

````
{{ $c := "<strong>blub <span style='color:red;'>red</span></strong>"}}
{{ partial "shortcodes/notice.html" (dict
"context" .
"style" "primary"
"title" "Hot shit"
"icon" "skull-crossbones"
"content" $c
)}}
````

Replace the `$c := ...` part with 

````
{{ $quizzes := .Params.quizzes }}
$c := "<ul>"
{{ range $quizzes }}
    {{ $name := index . "name" }}
    {{ $link := index . "link" }}
    $c = printf "%s%s" $c "<li><a href=\"" + $link "\">" + (default $link $name) + "</a></li>"
{{ end }}
$c = printf "%s%s" $c "</ul>"
````

I haven't tested the code and hope there are no typos.

**Reply by cagix** [reply]
thanks for the idea! couldn't wrap my head around the `print` function ...

just for the sake of completeness: had to wrap all lines in `{{ ... }}` and to rewrite the main `print` part, as `+` doesn't seem to concatenate strings? also removed escaping of `"` by using `'` to improve readability (although not entirely sure that's a good move). 

so here's the working example: 

```
{{ $quizzes := .Params.quizzes }}
{{ $c := "<ul>" }}
{{ range $quizzes }}
    {{ $name := index . "name" }}
    {{ $link := index . "link" }}
    {{ $c = printf "%s <li><a href='%s'>%s</a></li>" $c ($link | safeURL) (default $link $name) }}
{{ end }}
{{ $c = printf "%s%s" $c "</ul>" }}

{{ partial "shortcodes/notice.html" (dict
"context" .
"style" "tip"
"title" "Quizzes"
"content" $c
)}}
```

**Reply by McShelby** [reply]
> had to wrap all lines in `{{ ... }}`

Yeah, my fault. I knew I mess up if I don't test it ;-)

> as `+` doesn't seem to concatenate strings

Sadly not. You always have to use `printf`

> `"` by using `'` to improve readability (although not entirely sure that's a good move)

That's totally fine. It's just my personal preference to stay with one style in the whole project for consistency reasons.

**Reply by cagix** [reply]
thx for your quick response and your work! this is a great theme :)
